### DISCUSSION TITLE
What should happen when the chosen active project changes

### DISCUSSION THREAD

**Discussion by jeremypw** [elementary/code#1053] [General]
See https://github.com/elementary/code/pull/1030 and https://github.com/elementary/code/issues/991

With the above PR it will be possible to choose an active project from amongst those loaded using a headbar widget. If there are several projects, each with documents open what should happen when an active project is chosen?

Some possibilities
* Nothing other than the target for global search action, new branch action etc changes.
* The other projects are collapsed but their documents remain open
* The other projects are hidden but remain open (along with closing the associated documents)
* The other projects are closed along with the associated documents but remain in a list of recently used projects.

An related question is what should happen if the currently focused document is not part of the chosen project? At the moment, the active project is synced to the focused document - should this be discontinued?

If we go down the road of closing documents in inactive projects, it might be useful to be able to pin certain documents so that they remain open.

**Comment by Marukesu** [comment]
As #991 already says, what i would expect is that the sidebar only list files from the current project. but I don't think that changing the project should change what is opened.

About opened documents from other projects, I think we should only change the opened project if the user changed it in the popover or if it opened a new project folder.

**Comment by jeremypw** [comment]
> what i would expect is that the sidebar only list files from the current project.

There is more than one way this could be done:
 - Close other projects (which would also close their associated documents as things stand so this is probably not the desired solution)
 - Hide other projects in the sidebar, so the user cannot interact with them until they are made active again using the popover.
 - Collapse other projects in the sidebar

> I don't think that changing the project should change what is opened.

This implies that an inactive project should no longer become active when one of its documents is focused.  This raises the questions about what should happen when a keyboard shortcut to an action (e.g. a global search) is done - at the moment only the active project is acted on (which would not include the focused document) - is that desired?

**Reply by Marukesu** [reply]
> There is more than one way this could be done:

I would go with hiding the others projects.

> what should happen when a keyboard shortcut to an action (e.g. a global search) is done - at the moment only the active project is acted on (which would not include the focused document) - is that desired?

i believe so, assuming that the user is working on the active project, if the focused document is from a inactive project, i think it's for searching something in that specific file, so no need to interact with the inactive project other than with that file.
